Week 3: 구수민 과제 제출

tn-als opened the pull request:
Feat: 과제 1
- SignUp 페이지 구현
- ConstraintLayout에 적합한 방식으로 구현
- Layout 단위에 margin 적용하여 크기 제한
- id, password, 닉네임, mbti, about me를 입력받음

Feat: 과제 2
-계산기 구현
- 숫자 두 개 입력 후, 사칙연산 기호를 누르면 아래에 결과가 출력되는 프로그램
- 숫자 입력 시 숫자만 있는 키패드가 생성
- 단, /의 경우 두 번째 숫자가 0일 경우 "Result: -"를 출력하고, Toast로 "0으로 나눌 수 없습니다." 출력

tn-als committed:
Merge pull request #3 from KGU-C-Lab/master

Week2

tn-als committed:
Week 3: 구수민 과제 제출

Feat: 과제 1
- SignUp 페이지 구현
- ConstraintLayout에 적합한 방식으로 구현
- Layout 단위에 margin 적용하여 크기 제한
- id, password, 닉네임, mbti, about me를 입력받음

Feat: 과제 2
-계산기 구현
- 숫자 두 개 입력 후, 사칙연산 기호를 누르면 아래에 결과가 출력되는 프로그램
- 숫자 입력 시 숫자만 있는 키패드가 생성
- 단, /의 경우 두번째 숫자가 0일 경우 "Result: -"를 출력하고, Toast로 "0으로 나눌 수 없습니다." 출력

tn-als committed:
Merge pull request #4 from KGU-C-Lab/master

Week3

chattymin commented on 구수민/Week 3/HW_1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,141 @@
+        android:hint="비밀번호를 입력해주세요"
+        android:inputType="textPassword"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.491"
```
이 코드의 경우 xml의 view 상에서만 특정 물체가 이동한것 처럼 보이게 만드는 기능을 합니다.
그렇기 때문에 실제 결과물에 영향을 끼치지 않는 코드이기 때문에 지우는게 좋을 것 같네요

chattymin commented on 구수민/Week 3/HW_2/app/src/main/java/com/tnals/calculator/MainActivity.kt:
```diff
@@ -0,0 +1,71 @@
+        val divideButton = findViewById<Button>(R.id.divideButton)
+        val result = findViewById<TextView>(R.id.result)
+
+        fun sumNum(operator: String): String {
+            val num1 = numInput1.text.toString().toInt()
+            val num2 = numInput2.text.toString().toInt()
+            when (operator) {
+                "+" -> return (num1 + num2).toString()
+                "-" -> return (num1 - num2).toString()
+                "*" -> return (num1 * num2).toString()
+                "/" -> {
+                    if (num2 == 0)
+                        return "-"
+                    return (num1 / num2).toString()
+                }
+            }
+            return "Error"
+        }
```
함수화 한것 정말 좋네요!!

chattymin replied:
그런데 Int로 변환하기 전, 숫자인지 검사하는 과정이 있으면 더 좋을 것 같네용
예를 들어서 " " 이렇게 공백이 들어갈 경우는 에러가 발생합니다!

chattymin commented on 구수민/Week 3/HW_2/app/src/main/java/com/tnals/calculator/MainActivity.kt:
```diff
@@ -0,0 +1,71 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val numInput1 = findViewById<EditText>(R.id.inputNum1)
+        val numInput2 = findViewById<EditText>(R.id.inputNum2)
+        val plusButton = findViewById<Button>(R.id.plusButton)
+        val minusButton = findViewById<Button>(R.id.minusButton)
+        val multiplyButton = findViewById<Button>(R.id.multiplyButton)
+        val divideButton = findViewById<Button>(R.id.divideButton)
+        val result = findViewById<TextView>(R.id.result)
```
binding을 사용하는 방법으로 수정합시다 허허
findViewById는 안쓰는게 좋아용

chattymin commented on 구수민/Week 3/HW_2/app/src/main/java/com/tnals/calculator/MainActivity.kt:
```diff
@@ -0,0 +1,71 @@
+
+class MainActivity : AppCompatActivity() {
+
+    @SuppressLint("SetTextI18n")
```
에러를 무시하는 어노테이션은 최대한 쓰지 않는 것이 좋습니다.
물론 기능에 문제가 있는 어노테이션은 아니지만, 어노테이션 없이 해결하는 방법을 찾아보시는 것을 추천합니다:)

tn-als committed:
Wee 3: 구수민 과제 수정

Fis: 과제 1
- SignUp 페이지 구현
- 실제 결과물과 관련 없는 코드 지움

Fis: 과제 2
- 계신기 구현
- int로 변환한 숫자가 아니면 'error' 출력
- findViewById가 아닌 binding을 사용
- 어노테이션 삭제 및 strings.xml에 <string name="result">Result : %s</string> 추

chattymin commented on 구수민/Week 3/HW_2/app/src/main/java/com/tnals/calculator/MainActivity.kt:
```diff
@@ -0,0 +1,69 @@
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        fun sumNum(operator: String): String {
+            val num1: Int
+            val num2: Int
+
+            try {
+                num1 = binding.inputNum1.text.toString().toInt()
+                num2 = binding.inputNum2.text.toString().toInt()
+            } catch (ex: NumberFormatException) {
+                return "Error"
+            }
+
+            when (operator) {
+                "+" -> return (num1 + num2).toString()
+                "-" -> return (num1 - num2).toString()
+                "*" -> return (num1 * num2).toString()
+                "/" -> {
+                    if (num2 == 0)
+                        return "-"
+                    return (num1 / num2).toString()
+                }
+            }
+            return "Error"
+        }
+
```
에러 핸들링 폼 미쳐따...

chattymin commented on 구수민/Week 3/HW_2/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,103 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView" />
```
@+id/~ : 새로운 객체 생성
@id/~ : 기존 객체 활용

이런 뜻을 가지고 있답니다~
여기서는 기존에 있는 textView를 활용하는 것이기 때문에 +를 빼면 더 좋겠죠?

chattymin reviewed (approved):
감탄하며 이마 치다가 이마가 오목해졌습니다.
코드 퀄리티 장난아니네욥
